.github/workflows/format-code.yml

```
@@ -1,19 +1,71 @@
+---
 name: "Format Code: ensure code formatting guidelines are met"
+
 on:
-  pull_request:
-    types: [opened, edited, reopened, synchronize]
+  workflow_dispatch:
+  schedule:
+    - cron: "55 4 * * *"
+
+permissions:
+  contents: write
+  pull-requests: write
 
-permissions: {}
+env:
+  # Apply linter fixes configuration
+  APPLY_FIXES: all # When active, APPLY_FIXES must also be defined as environment variable (in github/workflows/mega-linter.yml or other CI tool)
+  APPLY_FIXES_EVENT: pull_request # Decide which event triggers application of fixes in a commit or a PR (pull_request, push, all)
+  APPLY_FIXES_MODE: pull_request # If APPLY_FIXES is used, defines if the fixes are directly committed (commit) or posted in a PR (pull_request)
+
+concurrency:
+  group: ${{ github.ref }}-${{ github.workflow }}
+  cancel-in-progress: true
 
 jobs:
-  format-code:
-    permissions:
-      contents: write
+  build:
+    name: MegaLinter
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@ac593985615ec2ede58e132d2e21d2b1cbd6127c # v3.3.0
-      - uses: ministryofjustice/github-actions/code-formatter@15729e214b8e626847cf4b4596c7635ddb6b1b0a # v10
+      - name: Checkout Code
+        uses: actions/checkout@ac593985615ec2ede58e132d2e21d2b1cbd6127c
         with:
-            ignore-files: "README.md"
+          token: ${{ secrets.GITHUB_TOKEN }}
+          fetch-depth: 0
+
+      - name: Prepare Git options
+        run: bash ./scripts/git-setup.sh
+
+      - name: Create new branch
+        run: |
+          date=$(date +%Y-%m-%d)
+          branch_name="date-$date"
+          git checkout -b $branch_name
+
+      # MegaLinter
+      - name: Run linter
+        id: ml
+        # You can override MegaLinter flavor used to have faster performances
+        # More info at https://megalinter.io/flavors/
+        uses: oxsecurity/megalinter/flavors/terraform@8fd433c675d27ceca5a61ace53177c1bbfbf7f49
```

You may want to add the release in a comment after the hash (dependabot will update this as well) to help easily see what version is running

```suggestion
        uses: oxsecurity/megalinter/flavors/terraform@8fd433c675d27ceca5a61ace53177c1bbfbf7f49 #v6.18.0
```